---
title: "Project02 - Group01"
author: "Eva, Tobi, Kathi, Laura"
date: "14 Juni 2019"
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## data loading
```{r}


wd = getwd()

NCI_TPW_gep_treated = readRDS(paste0(wd, "/Data/NCI_TPW_gep_treated.rds"))
NCI_TPW_gep_untreated = readRDS(paste0(wd, "/Data/NCI_TPW_gep_untreated.rds"))
Metadata = read.delim(paste0(wd, "/Data/NCI_TPW_metadata.tsv"), header = TRUE, sep = "\t", stringsAsFactors = TRUE)
Cellline_Annotation = read.delim(paste0(wd, "/Data/cellline_annotation.tsv"), header = TRUE, sep = "\t", stringsAsFactors = TRUE)
Drug_Annotation = read.delim(paste0(wd, "/Data/drug_annotation.tsv"), header = TRUE, sep = "\t", stringsAsFactors = TRUE)
CCLE_mutations = readRDS(paste0(wd, "/Data/CCLE_mutations.rds"))
CCLE_copynumber = readRDS(paste0(wd, "/Data/CCLE_copynumber.rds"))
CCLE_basalexpression = readRDS(paste0(wd, "/Data/CCLE_basalexpression.rds"))
NegLogGI50 = as.data.frame(readRDS(paste0(wd, "/Data/NegLogGI50.rds")))
Treated = data.frame(NCI_TPW_gep_treated)
Untreated = data.frame(NCI_TPW_gep_untreated)
```

## data scaling
```{r}
list = list(Treated,Untreated)
nlist = lapply(list,scale)
Treated = as.data.frame(nlist[[1]])
Untreated = as.data.frame(nlist[[2]])
Fold_Change = Treated - Untreated
Fold_Change <- data.frame(Fold_Change)
rm(NCI_TPW_gep_treated,NCI_TPW_gep_untreated,list,nlist)

```

# broad analysis
### installing packages
```{r}
library(rstudioapi)
library(cluster) 
```


**Boxplots** (already normalized)
```{r}
boxplot(Treated)
```

```{r}
boxplot(Untreated)
```

```{r}
#boxplot(Fold_Change)
```

```{r}
list = list(Treated,Untreated,Fold_Change)
nlist = lapply(list,scale)
Treated = as.data.frame(nlist[[1]])
Untreated = as.data.frame(nlist[[2]])
Fold_Change = as.data.frame(nlist[[3]])
```


```{r}
boxplot(Treated,  ylab = "Gene expression profile", main = "Teated genexpressionprofiles",xaxt = "n")
```

```{r}
boxplot(Untreated, ylab = "Gene expression profile", main = "Untreated genexpressionprofiles",xaxt = "n")
```

```{r}
boxplot(Fold_Change, ylab = "Gene expression profile", main = "Fold_Change genexpressionprofiles",xaxt = "n")
```

```{r}
boxplot(Treated[,1:10], ylab = "Gene expression profile", main = "First 10 reated genexpressionprofiles")
```


```{r couloured boxplots, creating a susbet of Metadata based on the amount of tested genes in Treated}
Treated1 = readRDS(paste0(wd, "/Data/NCI_TPW_gep_treated.rds"))
df = data.frame(t(Treated1))
df.data <- data.frame(sample = rownames(df))
adjustedMeda = subset(Metadata, sample %in% intersect(Metadata$sample, df.data$sample))
rm(df,df.data, Treated1)
palette(rainbow(15))
boxplot(Treated, border=adjustedMeda$drug,xlab= "different Drugs" ,ylab = "Gene expression profile", main = "Teated genexpressionprofiles",xaxt ="n")
```

**Densityplot**abline show the 3 quantiles (      25%       50%       75%    )
```{r}
plot(density(NCI_TPW_gep_treated), "Densityplot Treated vs Untreated")
lines(density(NCI_TPW_gep_untreated), col = "indianred2")
legend("topright", legend = c("treated", "untreated"), col = c("black", "indianred2"), pch = 20)
abline(v = quantile(NCI_TPW_gep_treated)[2:4], col = c("lightblue", "blue",  "orange"), lty = 2)
```

**k-means clustering**
```{r}
# Performing a k-means on Treated
#Determining the number of clusters
topVarTreated = apply(Treated, 1, var)
summary(topVarTreated)

# Using the most variable, thus informative genes
topVarTreated75 = Treated[topVarTreated > quantile(topVarTreated, probs = 0.75), ]
dim(topVarTreated75)

```

```{r}
km = kmeans(x = t(topVarTreated75), centers = 3, nstart = 10)
km$tot.withinss
```

```{r}
km = kmeans(x = t(topVarTreated75), centers = 2, nstart = 10)
km$tot.withinss
```

```{r}
#running a loop for the best n (searching for "ellbow")
wss = sapply(2:7, function(k) {
kmeans(x = t(topVarTreated75), centers = k)$tot.withinss})            
plot(2:7, wss, type = "b", pch = 19, xlab = "Number of clusters K", ylab = "Total within-clusters sum of squares", main = "Determining the amount of clusters from Treated")
```

```{r}
# Using the silhouett method
D = dist(t(topVarTreated75))
km = kmeans(x = t(topVarTreated75), centers = 10, nstart = 10)
s = silhouette(km$cluster, D)
plot(s)
```


**PCA**
```{r}
pca <- prcomp(t(Fold_Change), scale = TRUE)

```

```{r}
pca.data <- data.frame(sample = rownames(pca$x),
                       x = pca$x[,1],
                       y = pca$x[,2]
                       )
```

```{r}
# coloring 
metad.cl <- subset(Metadata, sample %in% intersect(Metadata$sample, pca.data$sample)) ## adjust row length of metadata to pca.data
metad.cl$mechanism <- Drug_Annotation$Mechanism[match(metad.cl$drug, Drug_Annotation$Drug)]
metad.cl$msi <- Cellline_Annotation$Microsatellite_instability_status[match(metad.cl$cell, Cellline_Annotation$Cell_Line_Name)]
```
```{r}
pca.var <- pca$sdev^2  # sdev calculates variation each PC accounts for
pca.var.per <- round(pca.var/sum(pca.var)*100, 1) 
# since percentages make more sense then normal variation values
# calculate % or variation, which is much more interesing
```
```{r}
plot(pca.var.per[1:8], type = "l", xlab = "Principal Components", ylab = "% variation")
```
```{r}
barplot(pca.var.per, main = "Scree plot", xlab = "Principal Components", ylab = "% variation")
```
```{r}
## get names of top 10 genes that contribute most to pc1
loading_scores_1 <- pca$rotation[,1]

gene_score <- abs(loading_scores_1) ## sort magnitude
gene_score_ranked <- sort(gene_score, decreasing = TRUE)

top_10_genes <- names(gene_score_ranked[1:10])
top_10_genes # show names of top 10 genes

```
```{r}
# plotting all important pcs agianst each other 
j<-1
while(j < 7){
  i <- 2
  while(i < 7){
    sapply(1:length(i), function(x){
      sapply(1:length(i), function(y){
        plot(x=pca$x[,j], y=pca$x[,i],  col = metad.cl$drug, xlab = j, ylab = i, main = "Colored by drug")
        })
      })
    i <- i+1}
  j <- j+1}

```


# specific analysis: lapatinib
###installing packages 
```{r}
library(cluster)
library(dplyr)
library(factoextra)
library(fpc)
library(dendextend)

install.packages("pheatmap")
install.packages("viridis")

library(pheatmap)
library(viridis)

```

#Analysis of the biomarker and t-test between treated and untreated cells


Expression of biomakers before and after the treatment of Lapatinib.
```{r Treated and Untreated}
boxplot(Untreated_t$ESR1, Treated_t$ESR1, 
        Untreated_t$ERBB2, Treated_t$ERBB2,
        Untreated_t$ERBB3, Treated_t$ERBB3, 
        Untreated_t$ERBB4, Treated_t$ERBB4,
        Untreated_t$KRT75, Treated_t$KRT75, 
        Untreated_t$PGR, Treated_t$PGR,
        col = viridis(12),
        names = c("ESR1 before", "ESR1 after", "ERBB2 before", "ERBB2 after", "ERBB3 before", "ERBB3 after", "ERBB4 before","ERBB4 after", "KRT75                       before", "KRT75 after", "PGR before",  "PGR after"),
        main="Expression of Biomakers before and after treatment with Lapatinib", 
        xlab="Biomarker",
        ylab="expression")
```


Basalexpresson of biomarkers for Lapatinib in breast cells, since Lapatinib is a drug against breast cancer.
```{r breast cells}
#biomarker matrix for breast cancer cell lines 
plot(expression_breast~Marker_Breast, data=df_breast_marker, col=viridis(6), main="Expression of Biomarkers in breast cells")
```


ggboxplot of der basalexpression of biomarkers in breast cells
```{r }
ggboxplot(df_breast_marker, x="Marker_Breast", y="expression_breast", color = "Marker_Breast",
          add = "jitter", legend = "none")+
          rotate_x_text(angle = 45)
          geom_hline(yintercept = mean(Biomarker_df$ERBB2), linetype = 2) # Add horizontal line at base mean
          #stat_compare_means(method = "anova", label.y = 12)+        # Add global annova p-value
          #stat_compare_means(label = "p.signif", method = "t.test",
          #ref.group = ".all.", hide.ns = TRUE)      # Pairwise comparison against all

```



expression of biomarker in breast cells
```{r basalexpression1}
pheatmap(data_biomarker_breast, 
                       show_rownames = TRUE, 
                       show_colnames = FALSE, 
                       cutree_cols = 4, 
                       cutree_rows = 3, 
                       color = viridis(12),
                       drop_levels = TRUE, 
                       clustering_method = "ward.D2", 
                       scale="row")
```




pheatmap of the basalexpression
```{r Basalexpression2}
library(pheatmap)
library(viridis)
pheatmap(CCLE_basalexpression,
         show_rownames = FALSE, 
         show_colnames = FALSE, 
         color = viridis(4),
         drop_levels = TRUE, 
         clustering_method = "ward.D2", 
         scale="row")
```



ggplot of all cell lines and the Biomarker
```{r }
library(ggpubr)
Biomarker_df<-data.frame(t(Matrix_biomarker))
Marker<-c(rep('ESR1',45), rep('ERBB2',45), rep('ERBB3',45), rep('ERBB4',45), rep('KRT75',45), rep('PGR',45))
expression<-c(Biomarker_df$`ESR1`, Biomarker$`ERBB2`, Biomarker$`ERBB3`, Biomarker$`ERBB4`, Biomarker$`KRT75`, Biomarker$`PGR`)
df_marker<-data.frame(expression, Marker)
plot(expression~Marker, data=df_marker)
ggboxplot(df_marker, x="Marker", y="expression", color = "Marker",
        add = "jitter", legend = "none")+
        rotate_x_text(angle = 45)+
        geom_hline(yintercept = mean(Biomarker_df$ERBB2), linetype = 2)+ # Add horizontal line at base mean
        stat_compare_means(method = "anova", label.y = 12)+        # Add global annova p-value
        stat_compare_means(label = "p.signif", method = "t.test",
                     ref.group = ".all.", hide.ns = TRUE)      # Pairwise comparison against all
```


pheatmap of the basalexpression of biomarkers in all cell lines 
```{r basalexpression3}
#pretty heatsmap
pheat_all_genes<-pheatmap(Matrix_biomarker, 
                          show_rownames = TRUE, 
                          show_colnames = FALSE, 
                          cutree_cols = 4, 
                          cutree_rows = 3, 
                          color = viridis(4),
                          drop_levels = TRUE, 
                          clustering_method = "ward.D2", 
                          scale="row")
```


setting a threshold to define overexpressed genes
```{r }
rmv.rows = apply(CCLE_basalexpression, 1, function(x) {
  sum(x<14)})
which(rmv.rows <14)
highest_expression = CCLE_basalexpression[-which(rmv.rows <14), ]  
rm(rmv.rows)
                                   
```


pheatmap of highly expressed genes
```{r}
library(viridis)
pheatmap(highest_expression,
         show_rownames = FALSE, 
         color=viridis(4),
         show_colnames = FALSE, 
         drop_levels = TRUE, 
         clustering_method = "ward.D2", 
         scale="row") 
```


```{r}
#searching for our own biomarkers
basalexpression <- data.frame(CCLE_basalexpression)
b<-apply(basalexpression,1, max)
highest_basalexpression<-sort(b, decreasing = TRUE)
head(highest_basalexpression)
```

boxplot of highly expressed genes before and after treatment with Lapatinib
```{r }
boxplot(Untreated_t$COX6C, Treated_t$COX6C, 
        Untreated_t$'2', Treated_t$'2',
        Untreated_t$RPS11, Treated_t$RPS11, 
        Untreated_t$GAPDH, Treated_t$GAPDH,
        Untreated_t$MT2A, Treated_t$MT2A, 
        Untreated_t$TUBA1B, Treated_t$TUBA1B,
        col = viridis(12),
        names = c("COX6C before", "COX6C after", 
                  "2 before", "2 after", 
                  "RPS11 before", "RPS11 after", 
                  "GAPDH before","GAPDH after", 
                  "$MT2A before", "$MT2A after", 
                  "TUBA1B before",  "TUBA1B after"),
        main="Expression of Biomakers before and after treatment with Lapatinib", 
        xlab="Biomarker",
        ylab="expression")
```


paired t-test between the Lapatinib treated und untreated data
```{r }
#genome wide paired t-test
before=as.matrix(LapatimibUnTreat)
after=as.matrix(LapatimibTreat)
t.test(before, after, paired=TRUE)
```

ggboxplot to compare the treated an untreated data
```{r }
my_data <- data.frame( 
  group = rep(c("before", "after")),
  expression = c(before,  after))


ggboxplot(my_data, x = "group", y = "expression", 
          color = "group", palette = c("#00AFBB", "#E7B800"),
          order = c("before", "after"),
          ylab = "expression", xlab = "group")
```


```{r}
#install.packages("matrixTests")
library(matrixTests)

#t-test over each column -> cell lines
col_t_paired(LapatimibUnTreat, LapatimibTreat, alternative = "two.sided", mu = 0,conf.level = 0.95)
```

#t-test over each row -> genes
```{r}
row_t_test<-row_t_paired(LapatimibUnTreat, LapatimibTreat, alternative = "two.sided", mu = 0,conf.level = 0.99)
summary(row_t_test$pvalue)
row_t_test
```
```{r}
row_t_test$pvalue
```





**k-means**
```{r}
LapatinibFold = select(Fold_Change, contains("Lapa"))

#Determining the number of clusters
topVarFold = apply(LapatinibFold, 1, var)
summary(topVarFold)

# Using the most variable, thus informative genes
topVarFold75 = LapatinibFold[topVarFold > quantile(topVarFold, probs = 0.75), ]
dim(topVarFold75)

km = kmeans(x = t(topVarFold75), centers = 2, nstart = 10)
km$tot.withinss

km = kmeans(x = t(topVarFold75), centers = 3, nstart = 10)
km$tot.withinss

#running a loop for the best n (searching for "ellbow")
wss = (sapply(2:7, function(k) {
  kmeans(x =t(topVarFold75), centers = k)$tot.withinss})  )            
plot(2:7, wss, type = "b", pch = 19, xlab = "Number of clusters K", ylab = "Total within-clusters sum of squares", main = "Determining the amount of clusters from Foldchange")
```

```{r}
# Using the silhouett method
D = dist(t(topVarFold75))
km = kmeans(x = t(topVarFold75), centers = 3, nstart = 10)
s = silhouette(km$cluster, D)
plot(s)
```

```{r}
#plot kmeans
clus <- kmeans(topVarFold75, centers=3)
clusplot(topVarFold75, clus$cluster, color=TRUE, shade=TRUE, labels=2, lines=0, col.clus = c("purple", "pink", "lightseagreen"), main = "K-means Plot", col.p = c("purple", "pink", "lightseagreen") )
plotcluster(topVarFold75, clus$cluster)
```

```{r}
#dendrogram
cor.mat = cor(topVarFold75, method = "spearman")
cor.dist = as.dist(1 - cor.mat)
cor.hc = hclust(cor.dist, method = "ward.D2")
cor.hc = as.dendrogram(cor.hc)
cor.hc %>% set("labels_col", value = c("purple", "pink", "lightseagreen"), k=3) %>%  plot(main = "Color labels \n by cluster", horiz = TRUE)


#PCA
pca = prcomp(topVarFold75, center = T, scale. = T)  
print(pca)
plot(pca, type = "l") #First two componets explain most of the variability in the data
plot(pca$rotation[, 1], pca$rotation[, 2], col = c("maroon1", "turquoise3"), pch = 19, xlab = "PC1", ylab = "PC2")
```

**PCA**
```{r}
L_fc <- select(Fold_Change, contains("Lapa"))

# PCA
pca <- prcomp(t(L_fc), scale = TRUE)
pca.var <- pca$sdev^2  # sdev calculates variation each PC accounts for
pca.var.per <- round(pca.var/sum(pca.var)*100, 1) 

```

```{r}
barplot(pca.var.per, main = "Scree plot", xlab = "Principal Components", ylab = "% variation")
```

```{r}
plot(pca.var.per[1:10], type = "l", xlab = "Principal Components", ylab = "% variation") # between 2 and 4 PCs
```

```{r}
pca.data <- data.frame(sample = rownames(pca$x),
                       x = pca$x[,1],
                       y = pca$x[,2]
)



## get names of top 10 genes that contribute most to pc1
loading_scores_1 <- pca$rotation[,1]

gene_score <- abs(loading_scores_1) ## sort magnitude
gene_score_ranked <- sort(gene_score, decreasing = TRUE)

top_10_genes <- names(gene_score_ranked[1:10])
top_10_genes # show names of top 10 genes

# coloring
metad.cl <- subset(Metadata, sample %in% intersect(Metadata$sample, pca.data$sample))  
metad.cl$msi <- Cellline_Annotation$Microsatellite_instability_status[match(metad.cl$cell, Cellline_Annotation$Cell_Line_Name)]
metad.cl$inoculation_d <- Cellline_Annotation$Inoculation_Density[match(metad.cl$cell, Cellline_Annotation$Cell_Line_Name)]
metad.cl$doubling_time <- Cellline_Annotation$Doubling_Time[match(metad.cl$cell, Cellline_Annotation$Cell_Line_Name)]
metad.cl$cancer_type <- Cellline_Annotation$Cancer_type[match(metad.cl$cell, Cellline_Annotation$Cell_Line_Name)]
```

```{r}
# plottting all importantn PCs
j<-1
while(j < 5){
  i <- 2
  while(i < 5){
    sapply(1:length(i), function(x){
      sapply(1:length(i), function(y){
        plot(x=pca$x[,j], y=pca$x[,i],  col = metad.cl$tissue, xlab = j, ylab = i, main = "Colored by tissue")
      })
    })
    i <- i+1}
  j <- j+1}
```

**pearson and spearman correlation**
```{r}
# Pearson+ Spearman correlation+plot

cor(Cellline_Annotation$Doubling_Time, Cellline_Annotation$Inoculation_Density, method = "pearson")
```

```{r}
# Spearman correlation
cor(Cellline_Annotation$Doubling_Time, Cellline_Annotation$Inoculation_Density, method = "spearman")
```

```{r}
plot(Cellline_Annotation$Doubling_Time, Cellline_Annotation$Inoculation_Density, pch= 16, col= "blue", main = "Pearson correlation between Doubling Time and Inoculation Density", xlab = "Doubling Time", ylab = "Inoculation Density")
lm(Cellline_Annotation$Inoculation_Density~ Cellline_Annotation$Doubling_Time)

#   Call:
#   lm(formula = Cellline_Annotation$Inoculation_Density ~ Cellline_Annotation$Doubling_Time)
#
#   Coefficients:
#   (Intercept)  Cellline_Annotation$Doubling_Time  
#       9110.9                              169.4   


abline(lm(Cellline_Annotation$Inoculation_Density ~ Cellline_Annotation$Doubling_Time), col = "red", lwd = 2)
```

#Main questions
##question 1 Pedicted GI50-values
```{r}
library(dplyr)
Fold_ChangeLap = select(Fold_Change, contains("Lapa"))
NegLogGI50Lap = NegLogGI50[9,]
means = colMeans(Fold_ChangeLap)
Fold_Changemeans = as.data.frame(t(means))

asdf = gsub(x = colnames (Fold_Changemeans), pattern = "_lapatinib_10000nM_24h", replacement = "")
colnames(Fold_Changemeans) = asdf

asdfa = gsub(x = asdf, pattern = "X7", replacement = "7")
colnames(Fold_Changemeans) = asdfa


asd = gsub(x = colnames (NegLogGI50Lap), pattern = "-", replacement = ".")
colnames(NegLogGI50Lap) = asd

c1 = rbind(asd,NegLogGI50Lap)
c2 = rbind(asdfa,Fold_Changemeans)

c1 = t(c1)
c2 = t(c2)

c1 =as.data.frame(c1)
c2 =as.data.frame(c2)



c3 = subset(c1, `1` %in% intersect(c1$`1`, c2$V1))
c4 = as.numeric(as.character(c3$lapatinib))
adjustedNeglogI50Lap = as.data.frame(c4)
#adjustedNeglogI50Lap = as.data.frame(t(adjustedNeglogI50Lap))


Fold_Changemeans = as.data.frame(t(Fold_Changemeans))

combined1 = cbind(adjustedNeglogI50Lap, Fold_Changemeans)

names1 = c( "NegLogI50Lap","Fold_Changemeans")
colnames(combined1) = names1
                      
lmFold = lm(NegLogI50Lap ~ Fold_Changemeans, data = combined1)

summary(lmFold)

qqnorm(lmFold$residuals, main = "Test for normaldistribution of residuals")
qqline(lmFold$residuals)

plot(combined1$NegLogI50Lap, lmFold$fitted.values, pch = 20, col = "blue", xlab = "Real values", 
     ylab = "Predicted values", main = "Real and predicted values ~ linear regression (Fold_Changemeans)")
abline(0, 1, col = "red")

cor(combined1$NegLogI50Lap,combined1$Fold_Changemeans)
```

```{r}
#Same wth doubling-time

NegLogGI50Lap = NegLogGI50[9,]

#Sort by Cellline-Name
df = arrange(Cellline_Annotation, Cell_Line_Name)
Doublingtime = cbind.data.frame (df$Cell_Line_Name, df$Doubling_Time)

c21 = as.data.frame(t(NegLogGI50Lap))

combined2 = cbind(c21, Doublingtime$`df$Doubling_Time`)
names2 = c( "NegLogI50Lap","Doubling_Time")
colnames(combined2) = names2

combined2 =na.omit(combined2)

lmDouble = lm(NegLogI50Lap ~ Doubling_Time, data = combined2)

summary(lmDouble)

qqnorm(lmDouble$residuals, main = "Test for normaldistribution of residuals")
qqline(lmDouble$residuals)

plot(combined2$NegLogI50Lap, lmDouble$fitted.values, pch = 20, col = "blue", xlab = "Real values", 
     ylab = "Predicted values", main = "Real and predicted values ~ linear regression (Doubling-Time)")
abline(0, 1, col = "red")

cor(combined2$NegLogI50Lap,combined2$Doubling_Time)
```

```{r}
#Multiple regression

qwer = gsub(x =Doublingtime$`df$Cell_Line_Name`, pattern = "-", replacement = ".")
Doublingtime1 =  rbind(qwer,Doublingtime$`df$Doubling_Time`)
Doublingtime1 = as.data.frame(t(Doublingtime1)) 

c31 = subset(Doublingtime1, qwer %in% intersect(Doublingtime1$qwer, c2$V1))
c41 = as.numeric(as.character(c31$V2))
adjustedDoubling_Time = as.data.frame(c41)

combined3 = cbind(adjustedNeglogI50Lap, Fold_Changemeans, adjustedDoubling_Time)
names3 = c( "NegLogI50Lap","Fold_Changemeans","Doubling_Time")
colnames(combined3) = names3

mlr = lm(NegLogI50Lap ~ ., data = combined3)

summary(mlr)

qqnorm(mlr$residuals, main = "Test for normaldistribution of residuals")
qqline(mlr$residuals)

plot(combined3$NegLogI50Lap, mlr$fitted.values, pch = 20, col = "blue", xlab = "Real values", 
     ylab = "Predicted values" , main = "Real and predicted values ~ multiple regression")
abline(0, 1, col = "red")
```

##question 2 erlotinib vs lapatinib
```{r}
#generell correlation
n= as.data.frame(t(NegLogGI50))
rmv.rows = apply(n, 1, function(x) {
  sum(is.na(x))
})
NLGI50.all = n[-which(rmv.rows > 0), ]  # Removing any row with 1 or more missing values
rm(rmv.rows, n, NegLogGI50)
cor.mat = as.data.frame(cor(NLGI50.all[, 1:ncol(NLGI50.all)], method = "pearson")) #Vergleich zu anderen
round(cor.mat, 2)
```

```{r}
par(mar = c(3, 3, 0.5, 0.5), mgp = c(1.5, 0.5, 0), las = 2)
pairs(NLGI50.all[, 1:ncol(NLGI50.all)], pch = 20, cex = 0.5, col = "cyan") #oder: plot(NLGI50.all, col= "magenta2", pch = 20, cex =0.75, main = "Correlation_NegLogGI50")

diff = data.frame(erlotinib = NLGI50.all$erlotinib - mean(NLGI50.all$erlotinib), lapatinib = NLGI50.all$lapatinib- mean(NLGI50.all$lapatinib))
diff$celllines = rownames(NLGI50.all)
```
*plot erlotinib all genes*
```{r}
ggplot(diff, aes(x=celllines,y=erlotinib))+geom_bar(stat="identity")+coord_flip() + labs(title="")
```
*plot lapatinib all genes*
```{r}
ggplot(diff, aes(x=celllines,y=lapatinib))+geom_bar(stat="identity")+coord_flip()
```
*correlation erlotinib , lapatinib*
```{r}
#correlation erlotinib, lapatinib gesamt
cor(NLGI50.all$erlotinib, NLGI50.all$lapatinib)
# 0.6528188
```

*Breast genes*
```{r}
#only breast with mean all
### load data
Metadata_Lapatinib_treated = Metadata[which(Metadata$drug == "lapatinib" & Metadata$dose != "0nM"),]
NegLogGI50 = as.data.frame(readRDS(paste0(wd, "/Data/NegLogGI50.rds")))
#breast genes from Metadata (421:474)
Breast_Metadata_L_treated = Metadata[which(Metadata$drug == "lapatinib" & Metadata$dose != "0nM" & Metadata$tissue == "Breast"),] 
celllines = Breast_Metadata_L_treated$cell
NegLogGI50.breast = as.data.frame(t(NegLogGI50[c("erlotinib", "lapatinib"), celllines]))

#Differenz
dif.NegLogGI50.breast = data.frame(erlotinib = NegLogGI50.breast$erlotinib -  mean(NLGI50.all$erlotinib), lapatinib = NegLogGI50.breast$lapatinib -  mean(NLGI50.all$lapatinib))
rownames(dif.NegLogGI50.breast) = rownames(NegLogGI50.breast)
dif.NegLogGI50.breast$cellline = rownames(dif.NegLogGI50.breast)

# PLot
library(ggplot2)

ggplot(dif.NegLogGI50.breast,aes(x=cellline,y=erlotinib))+geom_bar(stat="identity", fill="magenta2")+geom_text(aes(label=round(erlotinib, 2)), vjust=-0.5, color="black", size=3) +coord_flip() + labs(title="Breast genes Erlotinib")
```
*plot lapatinib*
```{r}
ggplot(dif.NegLogGI50.breast,aes(x=cellline,y=lapatinib))+geom_bar(stat="identity", fill="magenta2")+geom_text(aes(label=round(lapatinib, 2)), vjust=-0.5, color="black", size=3)+coord_flip() + labs(title="Breast genes Lapatinib")
```
*correlation breast*
```{r}
cor(NegLogGI50.breast$erlotinib, NegLogGI50.breast$lapatinib, method = "pearson")
#0.7416895
```

###anova

selection of Lapatinib and Erlotinib treated cells
```{r}
 install.packages("dplyr")
  library(dplyr)
  

lapa<-data.frame(Metadata[which(Metadata[,'drug'] == "lapatinib"), ])
erlo<-data.frame(Metadata[which(Metadata[,'drug'] == "erlotinib"), ])

el<-right_join(lapa,erlo, by="cell")
el

rmv.rows = apply(el, 1, function(x) {
  sum(is.na(x))
})  # Go through each row and sum up all missing values
row.names(rmv.rows)
```


```{r}
fc<-(Treated-Untreated)
fc<-data.frame(scale(fc))
all<-data.frame(fc[grep("lapatinib|erlotinib", colnames(fc))])
all

#since erlotinip contains more columns than lapatinib, we have to remove these columns
library(dplyr)
all<-data.frame(all %>% select(-"CCRF.CEM_erlotinib_0nM_24h", 
                    -"HL.60_erlotinib_0nM_24h", 
                    -"HT29_erlotinib_0nM_24h", 
                    -"K.562_erlotinib_0nM_24h", 
                    -"LOX_erlotinib_0nM_24h",
                    -"SR_erlotinib_0nM_24h",
                    -"COLO.205_lapatinib_0nM_24h"))
                   
```


Checking the rows
```{r}
la<-data.frame(all[grep("lapatinib", colnames(all))])
ncol(la)

er<-data.frame(all[grep("erlotinib", colnames(all))])
ncol(er)

erla<-data.frame(er,la)

ncol(all) #to prove if the columns are removed
```


Anova 
```{r}
#ggplot(lapatinib, erlotinib)

drug<-c(rep('Erlotinib',53), rep('Lapatinib',53))
expression_drug<-apply(erla, MARGIN = 2, sum)
df_drug<-data.frame(expression_drug, drug)

boxplot(erlotinib, lapatinib)
library(ggpubr)
ggboxplot (data = df_drug, x="drug", y="expression_drug", color = "drug",    
          add = "jitter", legend = "none")+
          rotate_x_text(angle = 45)+
          geom_hline(yintercept = mean(lapatinib$MCF7_lapatinib_0nM_24h), linetype = 2)+ # Add horizontal line at base mean
          stat_compare_means(method = "anova")+        # Add global annova p-value
          stat_compare_means(label = "p.signif", method = "t.test",
          ref.group = ".all.", hide.ns = TRUE)      # Pairwise comparison against all
```


## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r cars}
summary(cars)
```

## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
