---
title: "Project02 - Group01"
author: "Eva, Tobi, Kathi, Laura"
date: "24. July 2019"
output: html_document
---



```{r setup, include=FALSE, warning = FALSE}
knitr::opts_chunk$set(echo = TRUE,
                      chache = TRUE)
#library packages
library(calibrate)
library(cluster) 
library(compare)
library(data.table)
library(dendextend)
library(dplyr)
library(factoextra)
library(fpc)
library(ggplot2)
library(ggpubr)
library(ggrepel)
library(matrixTests)
library(pheatmap)
library(plotly)
library(RColorBrewer)
library(viridis)

```

# Table of content

[1. Introduction](#anchor1)  
  [1.1 Data loading](#css_id1)  
  [1.2 Data scaling](#css_id1.1)  

[2. Broad analysis](#anchor2)  
  [2.1 Boxplots](#css_id2.1)  
  [2.2 Densityplot](#css_id2.2)  
  [2.3 K-means clustering](#css_id2.3)  
  [2.4 PCA](#css_id2.4)  

[3. Specific analysis: lapatinib](#anchor3)

  [3.1 Analysis of biomarker and t-test between treated and untreated cells](#css_id3.1)  
    [3.1.1. Expression of biomakers before and after the treatment of Lapatinib](#css_class3.1.1)  
    [3.1.2 Searching for our own biomarkers](#css_class3.1.2)  
    [3.1.3 Boxplot of our biomarkers](#css_class3.1.3)  
    [3.1.4 ggBoxplot of our biomarkers' basalexpression in breast cells with Kruskal-Wallis test](#css_class3.1.4)  
    [3.1.5 Heatmap of biomarkers in breast cells](#css_class3.1.5)  

  [3.2. Paired t-test between the Lapatinib treated und untreated data](#css_id3.2)  
    [3.2.1 Genome wide paired t-test](#css_class3.2.1)  
    [3.2.2 Paired t-test of our biomarker before and after treatment](#css_class3.2.2)  
    [3.2.3 Paired t-test over genes, increased confidence level (99 %)](#css_class3.2.3)  
    [3.2.4 Visualization of the genome-wide paired t-test in a vulcano plot](#css_class3.2.4)  

  [3.3 K-means](#css_id3.3)
  
  [3.4 PCA](#css_id3.4)
  
  [3.5 Spearman correlation](#css_id3.5)

[4. Main questions](#anchor4)  
  [4.1 Question 1: Does the cell doubling time correlate with reduced drug sensitivity?](#css_id4.1)  
    [4.1.1 Linear regression ~ Foldchange-means](#css_class4.1.1)  
    [4.1.2 Linear regression ~ Foldchange-means biomarkers](#css_class4.1.2)  
    [4.1.3 Linear regression ~ Doubling-time](#css_class4.1.3)  
    [4.1.4 Multiple regression](#css_class4.1.4)  
    [4.1.5 Table of conclusion](#css_class4.1.5)  

  [4.2 Question 2: Does Lapatinib have a similar effect on lung cancer as Erlotinib?](#css_id4.2)  
    [4.2.1 Heatmap](#css_class4.2.1)  
    [4.2.2 Plot Erlotinib and Lapatinib, coloured by tissue](#css_class4.2.2)  
    [4.2.3 Pearson correlation](#css_class4.2.3)  
    [4.2.4 Lung cellines](#css_class4.2.4)  
    [4.2.5 Anova](#css_class4.2.5)  

  [4.3 Question 3: Comparing lapatinib treated breast and cns celllines](#css_id4.3)  
    [4.3.1 Volcano plot](#css_class4.3.1)
    [4.3.2 Heat map for visualization](#css_class4.3.2)
    [4.3.3 Investigation of correlation](#css_class4.3.3)
    
    
<div style = "text-align: justify">
# 1. Introduction {#anchor1}


Our project is about lapatinib and divided into three parts:
Broad analysis, specific analysis and our three main questions.
The main questions are the most important part of our project. They address whether cell doubling time correlates with reduced drug sensitivity and the effect of lapatinib on lung cancer and brain metastasis.
Twenty-five per cent of breast cancers overexpress ErbB2 / HER which leads to a more aggressive phenotype. Lapatinib blocks tyrosine kinases that belong to the HER2 receptor, which is increasingly found on the cell surface of cancer cells.
HER2 is a receptor for the growth factor epidermal growth factor (EGF), which stimulates the cell division of cancer cells. By blocking these HER2 receptors, the unregulated growth of cancer cells can be controlled again. Lapatinib binds the ATP-binding site of the receptor's intracellular domain and inhibits the survival and proliferation pathways of the tumor cell.


## 1.1 Data loading {#css_id1}
```{r data_loading, echo=FALSE}


wd = getwd()

NCI_TPW_gep_treated = readRDS(paste0(wd, "/Data/NCI_TPW_gep_treated.rds"))
NCI_TPW_gep_untreated = readRDS(paste0(wd, "/Data/NCI_TPW_gep_untreated.rds"))
Metadata = read.delim(paste0(wd, "/Data/NCI_TPW_metadata.tsv"), header = TRUE, sep = "\t", stringsAsFactors = TRUE)
Cellline_Annotation = read.delim(paste0(wd, "/Data/cellline_annotation.tsv"), header = TRUE, sep = "\t", stringsAsFactors = TRUE)
Drug_Annotation = read.delim(paste0(wd, "/Data/drug_annotation.tsv"), header = TRUE, sep = "\t", stringsAsFactors = TRUE)
CCLE_mutations = readRDS(paste0(wd, "/Data/CCLE_mutations.rds"))
CCLE_copynumber = readRDS(paste0(wd, "/Data/CCLE_copynumber.rds"))
CCLE_basalexpression = readRDS(paste0(wd, "/Data/CCLE_basalexpression.rds"))
NegLogGI50 = as.data.frame(readRDS(paste0(wd, "/Data/NegLogGI50.rds")))
Treated = data.frame(NCI_TPW_gep_treated)
Untreated = data.frame(NCI_TPW_gep_untreated)
```

## 1.2 Data scaling {#css_id1.1}

After checking for normalization, we scaled our data in the first place to provide the scaled data for further analysis.

```{r scaling}
list = list(Treated,Untreated)
nlist = lapply(list,scale)
Treated = as.data.frame(nlist[[1]])
Untreated = as.data.frame(nlist[[2]])
Fold_Change = Treated - Untreated
Fold_Change = data.frame(Fold_Change)
rm(NCI_TPW_gep_treated,NCI_TPW_gep_untreated,list,nlist)

```

### Remove missing values {#css_id1.2}
```{r remove_NAs}
n= as.data.frame(t(NegLogGI50))
rmv.rows = apply(n, 1, function(x) {
  sum(is.na(x))
})
NLGI50.all = n[-which(rmv.rows > 0), ]  # Removing any row with 1 or more missing values
rm(rmv.rows, n)
# used for second main question 
```


# 2. Broad analysis {#anchor2}

## 2.1 Boxplots {#css_id2.1}

As a very first step of our data analysis, we checked the for normalization.

```{r,echo=FALSE}
TreatedNS = readRDS(paste0(wd, "/Data/NCI_TPW_gep_treated.rds"))
boxplot(TreatedNS,  ylab = "Gene expression profile", main = "Treated genexpressionprofiles",xaxt = "n")

```

To check, if there is a connection between the different drugs, we coloured the plot concerning the drugs.

```{r coloured boxplots UNSCALED, creating a susbet of Metadata based on the amount of tested genes in Treated,echo=FALSE, fig.height = 5, fig.width = 7}
df = data.frame(t(TreatedNS))
df.data <- data.frame(sample = rownames(df))
adjustedMeda = subset(Metadata, sample %in% intersect(Metadata$sample, df.data$sample))
rm(df,df.data)

palette(magma(15))
par(mar=c(5, 4, 5, 9))
boxplot(TreatedNS, border=adjustedMeda$drug, 
        xlab= "Samples" ,
        ylab = "Gene expression profile",
        main = "Teated genexpressionprofiles ~ Unscaled",
        xaxt ="n" )
drugs = as.factor(levels(adjustedMeda$drug))

legend("topright", inset = c(-0.35,0), legend= drugs, xpd = TRUE, pch=19,
       col = drugs, title = "Drugs")

```

As shown, there is a connection between the different "boxes" and the different drugs. This could be the reason for different batches so we scale the data.

```{r coloured boxplots SCALED, creating a susbet of Metadata based on the amount of tested genes in Treated,echo=FALSE, fig.height = 5, fig.width = 7}
TreatedS = scale((Treated))
par(mar=c(5, 4, 5, 9))
boxplot(TreatedS, border=adjustedMeda$drug,xlab= "Samples" ,ylab = "Gene expression profile",
        main = "Teated genexpressionprofiles ~ Scaled",xaxt ="n" )
drugs = as.factor(levels(adjustedMeda$drug))

legend("topright", inset = c(-0.35,0), legend= drugs, xpd = TRUE, pch=19,
       col = drugs, title = "Drugs")

```

As you can see the scaling is necessary to compensate the batch effect. Scaling is then applied on Untreated as well before further analysis.



## 2.2 Densityplot {#css_id2.2}

As a other method to get in touch with the data, we checked if there is a difference between the treated and untreated genexpression. 
The abline shows the 3 quantiles (      25 %       50 %       75 %    )

```{r,echo=FALSE}
NCI_TPW_gep_treated = readRDS(paste0(wd, "/Data/NCI_TPW_gep_treated.rds"))
NCI_TPW_gep_untreated = readRDS(paste0(wd, "/Data/NCI_TPW_gep_untreated.rds"))
plot(density(NCI_TPW_gep_treated), "Densityplot Treated vs Untreated",xlab= "Expressionvalue")
lines(density(NCI_TPW_gep_untreated), col = "indianred2")
legend("topright", legend = c("Treated", "Untreated"), col = c("black", "indianred2"), pch = 20)
abline(v = quantile(NCI_TPW_gep_treated)[2:4], col = c("lightblue", "blue",  "orange"), lty = 2)
```

As assumed, the expression patterns don't differentiate much.



## 2.3 K-means clustering {#css_id2.3}

To look for clusters in the raw data we performed a k-menas clustering and searched for potentially clusters using the most variable (>75% Var), thus informative genes.

```{r, echo=FALSE}
topVarTreated = apply(Treated, 1, var)



topVarTreated75 = Treated[topVarTreated > quantile(topVarTreated, probs = 0.75), ]


```

Running a loop for the best cluster-amount (searching for an "ellbow")

```{r,echo=FALSE}
wss = sapply(1:7, function(k) {
kmeans(x = t(topVarTreated75), centers = k)$tot.withinss})            
plot(1:7, wss, type = "b", pch = 19, xlab = "Number of clusters K", ylab = "Total within-clusters sum of squares", main = "Determining the amount of clusters from Treated")
```

As we wanted an "ellbow", it is hard to determine the amount of clusters.



## 2.4 PCA {#css_id2.4}

Since the clustering method by performing k means clustering was not successful, we performed a principal component analysis for dimensionality reduction as another method for identifying clusters or patterns while preserving as much information of the original data set as possible.

```{r}
pca <- prcomp(t(Fold_Change), scale = TRUE)
```

By plotting a scree plot of the variation percentage, we can display how much variation a principal component captures from the original data. 

```{r}

pca.var <- pca$sdev^2  # sdev calculates variation each PC accounts for

pca.var.per <- round(pca.var/sum(pca.var)*100, 1) # since percentages make more sense than normal variation values calculate % of variation

plot(pca.var.per[1:10], main = "Scree plot", type = "b", xlab = "Principal Components", ylab = "% variation")
plot(cumsum(pca.var.per[1:15]), main = "cumulative variation", type = "l", xlab = "Principal Components", ylab = "% variation") 

```

As seen in the scree plot, the first three or four PCs have capture most of the information since the curve bends at an "elbow" at PC4 - this is our cutting off point.

```{r, echo=FALSE}
# creating data frame with all PCs 
# cleaning up sample names as they differ between matrices 
pca.data <- data.frame(pca$x)
rownames(pca.data) <- gsub(x = rownames(pca.data), pattern = "X786", replacement = "786")
pca.data <- cbind(sample =rownames(pca.data), pca.data)

```

PCs describe variation and account for the varied influences of the original characteristics. Such influences, or loadings, can be traced back from the PCA to find out which genes contribute most to the single PCS. 

```{r}
## get names of top 10 genes that contribute most to pc1
loading_scores_1 <- pca$rotation[,1]
gene_score <- abs(loading_scores_1) ## sort magnitude
gene_score_ranked <- sort(gene_score, decreasing = TRUE)


top_10_genes <- names(gene_score_ranked[1:10])
top_10_genes # show names of top 10 genes
```
```{r, echo=FALSE}
### Metadata color matrix for coloring 
Metadata$sample <- gsub(x = Metadata$sample, pattern = "-", replacement = ".")

metad.cl <- subset(Metadata, Metadata$sample %in% pca.data$sample) 
## adjust row length of metadata to pca.data


metad.cl$mechanism <- Drug_Annotation$Mechanism[match(metad.cl$drug, Drug_Annotation$Drug)]
metad.cl$msi <- Cellline_Annotation$Microsatellite_instability_status[match(metad.cl$cell, Cellline_Annotation$Cell_Line_Name)]

```

```{r, echo=FALSE}
# plotting all informative PCs
#color vectors for coloring by drug and tissue

#drug
magma <- magma(15)
color_drug = magma[metad.cl$drug]
drug <- levels(metad.cl$drug)

#tissue
viridis <- c("springgreen", "olivedrab4", "yellow2", "orange", "red2", "orchid1", "purple",  "turquoise2",   "navy")
color_tissue = viridis[metad.cl$tissue]
tissue <- levels(metad.cl$tissue)
```

To determine clusters of samples based on their similarity, we plotted PC1 and PC2 aswell as a PC2 and PC3 while colouring by tissue type and drug respectively.

```{r, PCA drug+tissue, fig.height=6, fig.width=12}
par(mfrow=c(1,2), cex=0.7)
#par(mar=c(4, 6, 3, 3))

## colored by drug
#plot PC1 and PC2
plot(pca$x[,1], 
     pca$x[,2], 
     col = color_drug,
     pch = 19, 
     xlab = paste("PC1 (",pca.var.per[1],"%)"), 
     ylab = paste("PC2 (",pca.var.per[2],"%)"))
#create legend
legend("topleft", 
       legend = drug, 
       col = magma, 
       pch = 19, 
       xpd = "TRUE",
       bty = "n",
       cex = 0.75
)
#create title
mtext("PCA of Fold Change  colored by drug", 
      side = 3, 
      line = -2,
      cex = 1.2,
      font = 2, 
      outer = TRUE)

#plot PC2 and PC3
plot(pca$x[,2], 
     pca$x[,3], 
     col = color_drug,
     pch = 19, 
     xlab = paste("PC2 (",pca.var.per[2],"%)"), 
     ylab = paste("PC3 (",pca.var.per[3],"%)"))
#create legend
legend("right", 
       legend = drug, 
       col = magma, 
       pch = 19, 
       xpd = "TRUE",
       bty = "n",
       cex = 0.75,
       inset = c(0, 2)
)
#create title
mtext("PCA of Fold Change  colored by drug", 
      side = 3, 
      line = -2,
      cex = 1.2,
      font = 2, 
      outer = TRUE)





## colored by tissue
#plot PC1 and PC2
par(mfrow=c(1,2), cex=0.7)

plot(pca$x[,1], 
     pca$x[,2], 
     col = color_tissue,
     pch = 19, 
     xlab = paste("PC1 (",pca.var.per[1],"%)"), 
     ylab = paste("PC2 (",pca.var.per[2],"%)"))
#create legend
legend("topleft", 
       legend = tissue, 
       col = viridis, 
       pch = 19, 
       xpd = "TRUE",
       bty = "n",
       cex = 0.75
)
#create title
mtext("PCA of Fold Change  colored by tissue", 
      side = 3, 
      line = -2,
      cex = 1.2,
      font = 2, 
      outer = TRUE)

#plot PC2 and PC3
plot(pca$x[,2], 
     pca$x[,3], 
     col = color_tissue,
     pch = 19, 
     xlab = paste("PC2 (",pca.var.per[2],"%)"), 
     ylab = paste("PC3 (",pca.var.per[3],"%)"))
#create legend
legend("right", 
       legend = tissue, 
       col = viridis, 
       pch = 19, 
       xpd = "TRUE",
       bty = "n",
       cex = 0.75,
       inset = c(0, 2)
)
#create title
mtext("PCA of Fold Change  colored by tissue", 
      side = 3, 
      line = -2,
      cex = 1.2,
      font = 2,
      outer = TRUE)

```

As we can see, in contrast to the k means method, clusters can be determined. Respecting the colour annotation, it is clear to say that the samples cluster according to drug treatment.   
</div>






<div style="text-align: justify">  
# 3. Specific Analysis: Lapatinib {#anchor3}

## 3.1 Analysis of biomarker and t-test between treated and untreated cells {#css_id3.1}


### 3.1.1. Expression of biomakers before and after the treatment of Lapatinib {#css_class3.1.1}
</div>
```{r}
Treated_t<-data.frame(t(Treated))
Untreated_t<-data.frame(t(Untreated))
LapatinibUntreated<-dplyr::select(Untreated, contains('Lapa'))
LapatinibTreated<-dplyr::select(Treated, contains('Lapa'))
before=as.matrix(LapatinibUntreated)
after=as.matrix(LapatinibTreated)
LapUn<-data.frame(t(before))
LapTreat<-data.frame(t(after))
```

<div style="text-align: justify">  
We found these genes as biomarkers in a paper and presented them in our first presentation. Unfortunately, we can not use them as a biomarkers.
(Source of the paper: Basal-Like Breast Cancer Defined by Five Biomarkers Has Superior Prognostic Value than Triple-Negative Phenotype, by Maggie C.U. Cheang, David Voduc, Chris Bajdik, et al. Clin Cancer Res 2008; 14:1368-1376)</div>
```{r Treated and Untreated, fig.height=8, fig.width=16, message=TRUE, paged.print=TRUE, echo=FALSE}
boxplot(LapUn$ESR1, LapTreat$ESR1, 
        LapUn$ERBB2, LapTreat$ERBB2,
        LapUn$ERBB3, LapTreat$ERBB3, 
        LapUn$ERBB4, LapTreat$ERBB4,
        LapUn$KRT75, LapTreat$KRT75, 
        LapUn$PGR, LapTreat$PGR,
        col=c("deeppink4", "deeppink4", "burlywood1", "burlywood1", "darkcyan", "darkcyan", "pink", "pink", "darkolivegreen3", "darkolivegreen3", "lightblue", "lightblue"),
        names = c("ESR1 before", "ESR1 after", "ERBB2 before", "ERBB2 after", "ERBB3 before", "ERBB3 after", "ERBB4 before","ERBB4 after", "KRT75 before", "KRT75 after", "PGR before",  "PGR   after"),
        main="Expression of biomakers before and after treatment with Lapatinib", 
        xlab="Biomarkers",
        ylab="Expression")
```
<div style="text-align: justify"> 

### 3.1.2 Searching for our own biomarkers {#css_class3.1.2}

The log2FoldChange was examined from the mean of all genes before and after treatment with lapatinib.</div>

```{r}
Treat<-colMeans(LapTreat, na.rm = TRUE)
Untreat<-colMeans(LapUn, na.rm = TRUE)
ut<-(Treat-Untreat)
ut_sort<-sort(ut, decreasing = TRUE)
head(ut_sort)
```
<div style="text-align: justify"> 
GDF3 Growth Differentiation Factor 3 belongs to the transforming growth factor beta (TGF-beta) superfamily and has some intrinsic activity.

NUPR1 stands for nuclear protein, transcriptional regulator, 1. It regulates the transcription during stress signals by empowering cells with resistance to stress.

DDIT3 encodes for the DNA damage-inducible transcript 3 protein, a member of the CCAAT/enhancer-binding protein which is a transcription factor. It is important for the response of cell stresses. The DNA damage-inducible transcript 3 protein induces cell cycle arrest and apoptosis in response to ER stress.

TRIB3 stands for Tribbles Pseudokinase 3. The kinase is induced by the transcription factor NF-kappaB. It is essential for cell proliferation.

ATF3 stands for activating transcription factor 3. It belongs to the mammalian activation transcription factor/cAMP responsive element-binding (CREB). ATF-3 is induced by physiological stress.

ASNS stands for Asparagine Synthetase: The enzyme catalyzes the production of the amino acid L-asparagine. 
</div>

<div style="text-align: justify"> 
### 3.1.3 Boxplot of our biomarkers {#css_class3.1.3}
</div>
```{r, fig.height=8, fig.width=18, message=TRUE, paged.print=TRUE}
boxplot(LapTreat$GDF15,LapUn$GDF15,
        LapTreat$NUPR1, LapUn$NUPR1, 
        LapTreat$DDIT3, LapUn$DDIT3, 
        LapTreat$TRIB3, LapUn$TRIB3,
        LapTreat$ATF3,LapUn$ATF3, 
        LapTreat$ASNS, LapUn$ASNS, 
        col=c("deeppink4", "deeppink4", "burlywood1", "burlywood1", "darkcyan", "darkcyan", "pink", "pink", "darkolivegreen3", "darkolivegreen3", "lightblue", "lightblue"),
        names = c("GDF15 before", "GDF15 after", "NUPR1 before", "NUPR1 after ", "DDIT3 before", "DDIT3 after" ,"TRIB3 before","TRIB3 after", "ATF3 before", "ATF3 after", "ASNS before",  "ASNS after"),
        main="Expression of Biomakers before and after treatment with Lapatinib", 
        xlab="Biomarkers",
        ylab="Expression")
```


```{r}
breastcells<-subset(Metadata, tissue == "Breast")
breast<-subset(CCLE_basalexpression, breastcells$cell  %in%  colnames(CCLE_basalexpression))
tbreastcells<-data.frame(t(breast))
breast_marker <- c(tbreastcells$GDF15,tbreastcells$NUPR1, tbreastcells$DDIT3, tbreastcells$TRIB3, tbreastcells$ATF3, tbreastcells$ASNS)
breast_marker_matrix <- data.frame(tbreastcells$GDF15,tbreastcells$NUPR1, tbreastcells$DDIT3, tbreastcells$TRIB3, tbreastcells$ATF3, tbreastcells$ASNS)
names(breast_marker_matrix)[names(breast_marker_matrix) == "tbreastcells.GDF15"] <- "GDF15"
names(breast_marker_matrix)[names(breast_marker_matrix) == "tbreastcells.NUPR1"] <- "NUPR1"
names(breast_marker_matrix)[names(breast_marker_matrix) == "tbreastcells.DDIT3"] <- "DDIT3"
names(breast_marker_matrix)[names(breast_marker_matrix) == "tbreastcells.TRIB3"] <- "TRIB3"
names(breast_marker_matrix)[names(breast_marker_matrix) == "tbreastcells.ATF3"] <- "ATF3"
names(breast_marker_matrix)[names(breast_marker_matrix) == "tbreastcells.ASNS"] <- "ASNS"
Biomarkers<-c(rep('DF15',45),rep('NUPR1',45),rep('DDIT3',45),rep('TRIB3',45),rep('ATF3',45),rep('ASNS',45))
Expression<-c(tbreastcells$GDF15,tbreastcells$NUPR1, tbreastcells$DDIT3, tbreastcells$TRIB3, tbreastcells$ATF3, tbreastcells$ASNS)
df_breast<-data.frame(Expression, Biomarkers)

```

<div style="text-align: justify"> 
### 3.1.4 ggBoxplot of our biomarkers' basalexpression in breast cells with Kruskal-Wallis test {#css_class3.1.4}

The null hypothesis is: there is no difference between the groups.
The Kruskal-Wallis test is for independent samples. It tests whether the central trends of several independent samples differ. </div>

```{r, fig.height=5, fig.width=8, message=TRUE}

ggboxplot(df_breast, x="Biomarkers", y="Expression", color = "Biomarkers",
          add = "jitter", legend = "none")+
  rotate_x_text(angle = 45)+
  stat_compare_means(method = "kruskal.test", label.y = 15)+   # Add global kurskal wallis test p-value
  stat_compare_means(label = "p.signif", method = "t.test",
                     ref.group = ".all.", hide.ns = TRUE)      # Pairwise comparison against all
```

### 3.1.5 Heatmap of biomarkers in breast cells {#css_class3.1.5}

```{r, fig.height=5, fig.width=8, message=TRUE}
pheat_breast<-pheatmap(breast_marker_matrix ,
                      show_rownames = FALSE, 
                      show_colnames =TRUE,
                      cutree_cols = 4, 
                      cutree_rows = 3, 
                      color = viridis(4),
                      drop_levels = TRUE, 
                      clustering_method = "ward.D2", 
                      scale="row")
pheat_breast
```
<div style="text-align: justify"> 

## 3.2. Paired t-test between the Lapatinib treated und untreated data {#css_id3.2}

### 3.2.1 Genome wide paired t-test {#css_class3.2.1}
</div>
```{r, echo=FALSE }
LapatinibUntreated<-dplyr::select(Untreated, contains("Lapa"))
LapatinibTreated<-dplyr::select(Treated, contains("Lapa"))
before=as.matrix(LapatinibUntreated)
after=as.matrix(LapatinibTreated)
ba<-data.frame(before, after)
```

```{r}
t.test(before, after, paired=TRUE)
```
<div style="text-align: justify"> 
### 3.2.2 Paired t-test of our biomarker before and after treatment {#css_class3.2.2}

The paired t-test shows a statistical significance of our biomarkers.</div>

```{r, table1}
before_biomarker<-LapUn %>% dplyr::select("GDF15", "NUPR1", "DDIT3", "TRIB3", "ATF3", "ASNS")
after_biomarker<-LapTreat %>% dplyr::select("GDF15", "NUPR1", "DDIT3", "TRIB3", "ATF3", "ASNS")
table<-data.table(col_t_paired(before_biomarker, after_biomarker, alternative = "two.sided", mu = 0,conf.level = 0.95))
Marker=c("GDF15", "NUPR1", "DDIT3", "TRIB3", "ATF3", "ASNS")
pvalue<-table$pvalue
table[, .(Marker,pvalue)]
```

<div style="text-align: justify"> 
### 3.2.3 Paired t-test over genes, increased confidence level (99 %) {#css_class3.2.3} 
</div>
```{r}
row_t_test<-as.data.frame(row_t_paired(before, after, alternative = "two.sided", mu = 0,conf.level = 0.99))
summary(row_t_test$pvalue)
```

<div style="text-align: justify"> 
### 3.2.4 Visualization of the genome-wide paired t-test in a vulcano plot {#css_class3.2.4}
</div>
```{r, fig.height=12, fig.width=12, message=TRUE, paged.print=TRUE}
Lap_t_test<-data.table(col_t_paired(LapUn, LapTreat, alternative = "two.sided", mu = 0,conf.level = 0.95))
padj<-p.adjust(Lap_t_test$pvalue, "BH")

Lapatinib_fc<-as.data.frame.numeric(colMeans(LapUn-LapTreat))
Lgenes<-(colnames(LapUn))

vulcano<-data.frame(genes=Lgenes,Fold=Lapatinib_fc, pvalue=padj)
names(vulcano)[names(vulcano) == "colMeans.LapUn...LapTreat."] <- "log2FoldChange"
with(vulcano, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-2.5,2)))

# Add colored points: red if padj<0.05, blue if log2FC>1, green if both)
with(subset(vulcano, padj<.05 ), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
with(subset(vulcano, abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(vulcano, padj<.05 & abs(log2FoldChange)>1), points(log2FoldChange, -log10(pvalue), pch=20, col="green"))

# Label points with the textxy function from the calibrate plot
with(subset(vulcano, padj<.05 & abs(log2FoldChange)>1), textxy(log2FoldChange, -log10(pvalue), labs = genes, cex=.8))
```



## 3.3 K-means {#css_id3.3}
```{r}
LapatinibFold = select(Fold_Change, contains("Lapa"))

#Determining the number of clusters
topVarFold = apply(LapatinibFold, 1, var)
summary(topVarFold)

# Using the most variable, thus informative genes
topVarFold75 = LapatinibFold[topVarFold > quantile(topVarFold, probs = 0.75), ]
dim(topVarFold75)

km = kmeans(x = t(topVarFold75), centers = 2, nstart = 10)
km$tot.withinss

km = kmeans(x = t(topVarFold75), centers = 3, nstart = 10)
km$tot.withinss

#running a loop for the best n (searching for "ellbow")
wss = (sapply(2:7, function(k) {
  kmeans(x =t(topVarFold75), centers = k)$tot.withinss})  )            
plot(2:7, wss, type = "b", pch = 19, xlab = "Number of clusters K", ylab = "Total within-clusters sum of squares", main = "Determining the amount of clusters from Foldchange")
```

```{r}
# Using the silhouett method
D = dist(t(topVarFold75))
km = kmeans(x = t(topVarFold75), centers = 3, nstart = 10)
s = silhouette(km$cluster, D)
plot(s)
```


## 3.4 PCA {#css_id3.4}

Since the clustering method by performing k means clustering again was not successful, we performed a principal component analysis over the selected lapatinib samples for identifying clusters or patterns while preserving as much information of the original data set as possible.

```{r,}
L_fc <- select(Fold_Change, contains("Lapa"))

# PCA
pca <- prcomp(t(L_fc), scale = TRUE)

```

By plotting a scree plot of the variation percentage, we can display how much variation a principal component captures from the original data.

```{r}
pca.var <- pca$sdev^2  # sdev calculates variation each PC accounts for
pca.var.per <- round(pca.var/sum(pca.var)*100, 1) 
# since percentages make more sense then normal variation values
# calculate % or variation, which is much more interesing


plot(pca.var.per[1:10], main = "Scree plot", type = "l", xlab = "Principal Components", ylab = "% variation")
plot(cumsum(pca.var.per[1:15]), main = "cumulative variation", type = "l", xlab = "Principal Components", ylab = "% variation") 
```
```{r, echo=FALSE}
pca.data <- data.frame(pca$x)
rownames(pca.data) <- gsub(x = rownames(pca.data), pattern = "X786", replacement = "786")
pca.data <- cbind(sample =rownames(pca.data), pca.data)

```


```{r}
## get names of top 10 genes that contribute most to pc1
loading_scores_1 <- pca$rotation[,1]

gene_score <- abs(loading_scores_1) ## sort magnitude
gene_score_ranked <- sort(gene_score, decreasing = TRUE)

top_10_genes <- names(gene_score_ranked[1:10])
top_10_genes # show names of top 10 genes

```
```{r, echo=FALSE}
### Metadata matrix for coloring 
Metadata$sample <- gsub(x = Metadata$sample, pattern = "-", replacement = ".")
metad.cl <- subset(Metadata, sample %in% intersect(Metadata$sample, pca.data$sample))  
metad.cl$msi <- Cellline_Annotation$Microsatellite_instability_status[match(metad.cl$cell, Cellline_Annotation$Cell_Line_Name)]
metad.cl$inoculation_d <- Cellline_Annotation$Inoculation_Density[match(metad.cl$cell, Cellline_Annotation$Cell_Line_Name)]
metad.cl$doubling_time <- Cellline_Annotation$Doubling_Time[match(metad.cl$cell, Cellline_Annotation$Cell_Line_Name)]
metad.cl$cancer_type <- Cellline_Annotation$Cancer_type[match(metad.cl$cell, Cellline_Annotation$Cell_Line_Name)]

```
```{r, echo=FALSE}
#color vectors for coloring by msi and tissue
colormsi <- brewer.pal(3, "Set1")
color_msi = colormsi[metad.cl$msi]
msi <- levels(metad.cl$msi)

magma <- c("springgreen", "olivedrab4", "yellow2", "orange", "red2", "orchid1", "purple",  "turquoise2",   "navy")
color_tissue = magma[metad.cl$tissue]
tissue <- levels(metad.cl$tissue)

```

To determine clusters of samples based on their similarity, we plotted PC1 and PC2 aswell as a PC2 and PC3 while colouring by tissue type and drug respectively.

```{r, fig.height = 4.5, fig.width = 7.2}
par(mfrow=c(2,2), cex=0.7, oma=c(2,3,2,2))
par(mar=c(4, 6, 3, 3))
## colored by msi
#plot PC1 and PC2
msi12 <- plot(pca$x[,1], 
          pca$x[,2], 
          col = color_msi,
          pch = 19, 
          xlab = paste("PC1 (",pca.var.per[1],"%)"), 
          ylab = paste("PC2 (",pca.var.per[2],"%)"))

#create legend
legend(450, 0, 
       legend = msi, 
       col = colormsi, 
       pch = 19, 
       xpd = "FALSE",
       bty = "n",
       cex = 0.75
)
#create title
mtext("PCA of Fold Change colored by MSI", 
      side = 3, 
      line = -2,
      cex = 1.2,
      font = 2,
      outer = TRUE)


#plot PC2 and PC3
msi23 <- plot(pca$x[,2], 
          pca$x[,3], 
          col = color_msi,
          pch = 19, 
          xlab = paste("PC2 (",pca.var.per[2],"%)"), 
          ylab = paste("PC3 (",pca.var.per[3],"%)"))
    



##colored by tissue
#plot PC1 and PC2
tissue12 <- plot(pca$x[,1], 
              pca$x[,2], 
              col = color_tissue,
              pch = 19, 
              xlab = paste("PC1 (",pca.var.per[1],"%)"), 
              ylab = paste("PC2 (",pca.var.per[2],"%)"))
             
#create legend
legend(450, 0, 
       legend = tissue, 
       col = magma, 
       pch = 19, 
       xpd = "TRUE",
       bty = "n",
       cex = 0.75
)
#create title
mtext("PCA of Fold Change colored by tissue", 
      side = 3, 
      line = -16,
      cex = 1.2,
      font = 2,
      outer = TRUE)

#plot PC2 and PC3
tissue23 <- plot(pca$x[,2], 
              pca$x[,3], 
              col = color_tissue,
              pch = 19, 
              xlab = paste("PC2 (",pca.var.per[2],"%)"), 
              ylab = paste("PC3 (",pca.var.per[3],"%)"))


```

As we can see, rarely any clusters can be determined. In Addition, the colour annotation does not reveal any pattern of clustering.


## 3.5 Spearman correlation {#css_id3.5}

Modelling of a linear relationship between cell doubling time and inoculation density.


```{r}
# Spearman correlation
cor(Cellline_Annotation$Doubling_Time, Cellline_Annotation$Inoculation_Density, method = "spearman")
```


```{r}
plot(Cellline_Annotation$Doubling_Time, Cellline_Annotation$Inoculation_Density, pch= 16, col= "blue", main = "Spearman correlation between Doubling Time and Inoculation Density", xlab = "Doubling Time", ylab = "Inoculation Density")
lm(Cellline_Annotation$Inoculation_Density~ Cellline_Annotation$Doubling_Time)

abline(lm(Cellline_Annotation$Inoculation_Density ~ Cellline_Annotation$Doubling_Time), col = "red", lwd = 2)
```

As you can see from the plot, there is a positive correlation between inoculation density and doubling time. This means that the higher the inoculation density, the higher the doubling time, which suggests that the tumour grows slowly because the cells need more time to divide.



# 4. Main questions {#anchor4}


## 4.1 Question 1: Does the cell doubling time correlate with reduced drug sensitivity? {#css_id4.1}

<div style = "text-align: justify">
As mentioned in our presentation, we want to create a model to predict GI50-values and thus to predict, if Lapatinib is a good choice. Therefore, we took several datasets and performed 4 regression analyses.


### 4.1.1 Linear regression ~ Foldchange-means {#css_class4.1.1}

The first linear model tries to predict the G-50 value under the data of the Foldchange-means.


```{r, echo = FALSE}

NegLogGI50Lap = NegLogGI50[9,]
Fold_ChangeLap = select(Fold_Change, contains("Lapa"))
Fold_ChangeLap_t = t(Fold_ChangeLap)
means = colMeans(Fold_ChangeLap)
Fold_Changemeans = as.data.frame(t(means))

a2 = gsub(x = colnames (Fold_Changemeans), pattern = "_lapatinib_10000nM_24h", replacement = "")
colnames(Fold_Changemeans) = a2

a3 = gsub(x = a2, pattern = "X7", replacement = "7")
colnames(Fold_Changemeans) = a3


a1 = gsub(x = colnames (NegLogGI50Lap), pattern = "-", replacement = ".")
colnames(NegLogGI50Lap) = a1

c1 = rbind(a1,NegLogGI50Lap)
c2 = rbind(a3,Fold_Changemeans)

c1 = t(c1)
c2 = t(c2)

c1 =as.data.frame(c1)
c2 =as.data.frame(c2)


c3 = subset(c1, `1` %in% intersect(c1$`1`, c2$V1))
c4 = as.numeric(as.character(c3$lapatinib))
adjustedNeglogGI50Lap = as.data.frame(c4)



Fold_Changemeans = as.data.frame(t(Fold_Changemeans))

combined1 = cbind(adjustedNeglogGI50Lap, Fold_Changemeans)

names1 = c( "NegLogGI50Lap","Fold_Changemeans")
colnames(combined1) = names1
                      
lmFold = lm(NegLogGI50Lap ~ Fold_Changemeans, data = combined1)

#cor(combined1$NegLogI50Lap,combined1$Fold_Changemeans)

n = nrow(combined1)
rmse1 = sqrt(1/n * sum(lmFold$residuals^2))
#rmse1

#Split the data (Training - Testing)

i1.train = sample(1:nrow(combined1), 44)

dat1.train = combined1[i1.train, ]
dat1.test = combined1[-i1.train, ]

l1.train = lm(NegLogGI50Lap ~ Fold_Changemeans, data = dat1.train)
#summary(l1.train)

qqnorm(l1.train$residuals, main = "Test for normaldistribution of residuals")
qqline(l1.train$residuals)

plot(dat1.train$NegLogGI50Lap, l1.train$fitted.values, pch = 20, col = "blue", xlab = "Real values", 
     ylab = "Predicted values", main = "Linear regression ~ Fold_Changemeans ~ Training")
abline(0, 1, col = "red")

n = nrow(dat1.train)
rmse1.train = sqrt(1/n * sum(l1.train$residuals^2))
#rmse1.train

pred1 = predict(l1.train, newdata = dat1.test)

n = nrow(dat1.test)
residuals = dat1.test$NegLogGI50Lap - pred1
rmse1.test = sqrt(1/n * sum(residuals^2))
#rmse1.test

```

### 4.1.2 Linear regression ~ Foldchange-means biomarkers {#css_class4.1.2}

Linear regression is not suitable here, as the mean fold-change per cell line is not a meaningful indicator in a biological context.
Hence, we selected the top 20 genes to calculate our mean values. Afterwards linear regression was performed.

```{r, echo = FALSE}
Treated_t<-data.frame(t(Treated))
Untreated_t<-data.frame(t(Untreated))
LapatinibUntreated<-select(Untreated, contains("Lapa"))
LapatinibTreated<-select(Treated, contains("Lapa"))
LapTreat=as.data.frame(t(LapatinibUntreated))
LapUn=as.data.frame(t(LapatinibTreated))

m1<-colMeans(LapTreat, na.rm = TRUE)
m2<-colMeans(LapUn, na.rm = TRUE)

ut<-(m1-m2)
utabs = abs(ut)
ut_sort<-sort(utabs, decreasing = TRUE)
biomark= as.data.frame(ut_sort[1:20])
n1 = rownames(biomark)
biomark = cbind(n1,biomark)


Fold_ChangeLap = select(Fold_Change, contains("Lapa"))
Fold_ChangeLap_t = as.data.frame(Fold_ChangeLap)
Fold_ChangeLapBM = subset(Fold_ChangeLap_t, rownames(Fold_ChangeLap_t) %in% biomark$n1)

NegLogGI50Lap = NegLogGI50[9,]
means = colMeans(Fold_ChangeLapBM)
Fold_ChangemeansBM = as.data.frame(t(means))

a2 = gsub(x = colnames (Fold_ChangemeansBM), pattern = "_lapatinib_10000nM_24h", replacement = "")
colnames(Fold_ChangemeansBM) = a2

a3 = gsub(x = a2, pattern = "X7", replacement = "7")
colnames(Fold_ChangemeansBM) = a3


a1 = gsub(x = colnames (NegLogGI50Lap), pattern = "-", replacement = ".")
colnames(NegLogGI50Lap) = a1

c1 = rbind(a1,NegLogGI50Lap)
c2 = rbind(a3,Fold_ChangemeansBM)

c1 = t(c1)
c2 = t(c2)

c1 =as.data.frame(c1)
c2 =as.data.frame(c2)


c3 = subset(c1, `1` %in% intersect(c1$`1`, c2$V1))
c4 = as.numeric(as.character(c3$lapatinib))
adjustedNeglogGI50Lap = as.data.frame(c4)



Fold_ChangemeansBM = as.data.frame(t(Fold_ChangemeansBM))

combined11 = cbind(adjustedNeglogGI50Lap, Fold_ChangemeansBM)

names11 = c( "NegLogGI50Lap","Fold_ChangemeansBM")
colnames(combined11) = names11
                      
lmFold1 = lm(NegLogGI50Lap ~ Fold_ChangemeansBM, data = combined11)

#cor(combined1$NegLogI50Lap,combined1$Fold_Changemeans)

n = nrow(combined11)
rmse11 = sqrt(1/n * sum(lmFold1$residuals^2))
#rmse11

#Split the data (Training - Testing)

i11.train = sample(1:nrow(combined11), 44)

dat11.train = combined11[i11.train, ]
dat11.test = combined11[-i11.train, ]

l11.train = lm(NegLogGI50Lap ~ Fold_ChangemeansBM, data = dat11.train)
#summary(l11.train)

qqnorm(l11.train$residuals, main = "Test for normaldistribution of residuals")
qqline(l11.train$residuals)

plot(dat11.train$NegLogGI50Lap, l11.train$fitted.values, pch = 20, col = "blue", xlab = "Real values", 
     ylab = "Predicted values", main = "Linear regression ~ Fold_ChangemeansBM ~ Training")
abline(0, 1, col = "red")

n = nrow(dat11.train)
rmse11.train = sqrt(1/n * sum(l11.train$residuals^2))
#rmse11.train

pred11 = predict(l11.train, newdata = dat11.test)

n = nrow(dat11.test)
residuals = dat11.test$NegLogGI50Lap - pred11
rmse11.test = sqrt(1/n * sum(residuals^2))
#rmse11.test
```


### 4.1.3 Linear regression ~ Doubling-time {#css_class4.1.3}

The second linear model tries to predict the G-50 value under the data of the Doubling-time.
```{r,echo = FALSE}
NegLogGI50Lap = NegLogGI50[9,]

#Sort by Cellline-Name
df = arrange(Cellline_Annotation, Cell_Line_Name)
Doublingtime = cbind.data.frame (df$Cell_Line_Name, df$Doubling_Time)

c21 = as.data.frame(t(NegLogGI50Lap))

combined2 = cbind(c21, Doublingtime$`df$Doubling_Time`)
names2 = c( "NegLogGI50Lap","Doubling_Time")
colnames(combined2) = names2

combined2 =na.omit(combined2)

lmDouble = lm(NegLogGI50Lap ~ Doubling_Time, data = combined2)

#summary(lmDouble)

#cor(combined2$NegLogI50Lap,combined2$Doubling_Time)



#Split the data (Training - Testing)

n = nrow(combined2)
rmse2 = sqrt(1/n * sum(lmDouble$residuals^2))
#rmse2

i2.train = sample(1:nrow(combined2), 48)

dat2.train = combined2[i2.train, ]
dat2.test = combined2[-i2.train, ]

l2.train = lm(NegLogGI50Lap ~ Doubling_Time, data = dat2.train)
#summary(l2.train)

qqnorm(l2.train$residuals, main = "Test for normaldistribution of residuals")
qqline(l2.train$residuals)

plot(dat2.train$NegLogGI50Lap, l2.train$fitted.values, pch = 20, col = "blue", xlab = "Real values", 
     ylab = "Predicted values", main = "Linear regression ~ Doubling-Time ~ Training")
abline(0, 1, col = "red")

n = nrow(dat2.train)
rmse2.train = sqrt(1/n * sum(l2.train$residuals^2))
#rmse2.train

pred2 = predict(l2.train, newdata = dat2.test)

n = nrow(dat1.test)
residuals = dat2.test$NegLogGI50Lap - pred2
rmse2.test = sqrt(1/n * sum(residuals^2))
#rmse2.test
```


### 4.1.4 Multiple regression {#css_class4.1.4}

Finally, we did a multiple regression with both datasets to predict GI50-values.
```{r,echo = FALSE}
b1 = gsub(x =Doublingtime$`df$Cell_Line_Name`, pattern = "-", replacement = ".")
Doublingtime1 =  rbind(b1,Doublingtime$`df$Doubling_Time`)
Doublingtime1 = as.data.frame(t(Doublingtime1)) 

c31 = subset(Doublingtime1, b1 %in% intersect(Doublingtime1$b1, c2$V1))
c41 = as.numeric(as.character(c31$V2))
adjustedDoubling_Time = as.data.frame(c41)

combined3 = cbind(adjustedNeglogGI50Lap, Fold_ChangemeansBM, adjustedDoubling_Time)
names3 = c( "NegLogGI50Lap","Fold_ChangemeansBM","Doubling_Time")
colnames(combined3) = names3

mlr = lm(NegLogGI50Lap ~ ., data = combined3)

#summary(mlr)


#Split the data (Training - Testing)

n = nrow(combined3)
rmse3 = sqrt(1/n * sum(mlr$residuals^2))
#rmse3

i3.train = sample(1:nrow(combined3), 44)

dat3.train = combined3[i3.train, ]
dat3.test = combined3[-i3.train, ]

l3.train = lm(NegLogGI50Lap ~ ., data = dat3.train)
#summary(l3.train)

qqnorm(l3.train$residuals, main = "Test for normaldistribution of residuals")
qqline(l3.train$residuals)

plot(dat3.train$NegLogGI50Lap, l3.train$fitted.values, pch = 20, col = "blue", xlab = "Real values", 
     ylab = "Predicted values", main = "Multiple regression ~ Training")
abline(0, 1, col = "red")

n = nrow(dat3.train)
rmse3.train = sqrt(1/n * sum(l3.train$residuals^2))
#rmse3.train

pred3 = predict(l3.train, newdata = dat3.test)

n = nrow(dat3.test)
residuals = dat3.test$NegLogGI50Lap - pred3
rmse3.test = sqrt(1/n * sum(residuals^2))
#rmse3.test
```


### 4.1.5 Table of conclusion {#css_class4.1.5}

The following table shows important values. 

```{r, results='asis',echo = FALSE}
Rsrd =  c(summary(l1.train)$r.squared, summary(l11.train)$r.squared, summary(l2.train)$r.squared, summary(l3.train)$r.squared)
F_Stat = c(summary(l1.train)$fstatistic[1],summary(l11.train)$fstatistic[1],
           summary(l2.train)$fstatistic[1],summary(l3.train)$fstatistic[1])
Sum_RMSE = c(rmse1.test,rmse11.test,rmse2.test,rmse3.test)
Rsrd1 = as.data.frame(t(Rsrd))
table = rbind(Rsrd1,F_Stat,Sum_RMSE)
colnames(table) = c("Foldchange","FoldchangeBM","Doublingtime","Multiple regression")
rownames(table) = c("R-squared-value","F-statistic","RMSE of test-dataset")
knitr::kable(table, format = "markdown")

```


The data is hard to discuss, because we used random values for our test and training dataset. Hence, different values on every run are obtained. We could use "selected values" (First 20) but this would be kind of cheating.


## 4.2 Question 2: Does Lapatinib have a similar effect on lung cancer as Erlotinib? {#css_id4.2}

As we read in the paper "Antitumor and antiangiogenic effect of the dual EGFR and HER-2 tyrosine kinase inhibitor lapatinib in a lung cancer model. (Diaz et al., 2010)", lapatinib and erlotinib are said to have similar effects. To verify this, we first looked at the correlation of the drugs and then created a graph showing the difference between GI50 for a certain cell line and the middle GI50 for both drugs.


### 4.2.1 Heatmap {#css_class4.2.1}
```{r correlation heatmap}
cor = cor(NLGI50.all)
pheatmap(cor, col = cm.colors(256), main = "Correlation NLGI50")

```

Erlotinib and lapatinib have a strong correlation.


### 4.2.2 Plot Erlotinib and Lapatinib, coloured by tissue {#css_class4.2.2}
```{r plot erlotinib, fig.height=12, fig.width=16}

#differece
diff = data.frame(erlotinib = NLGI50.all$erlotinib - mean(NLGI50.all$erlotinib), lapatinib = NLGI50.all$lapatinib- mean(NLGI50.all$lapatinib))
diff$celllines = rownames(NLGI50.all)
#create vector to insert column tissue from Metadata

tissue = sapply(1:nrow(diff), function(x) {
  position = which(as.character(Metadata$cell) == diff[x, "celllines"])[1] #if tissue occurs several times, take the first
  out = as.character(Metadata[position, "tissue"]) #output the tissue at this position
  return(out)
})
diff$tissue = tissue
rm(tissue)

diff$celllines = factor(diff$celllines, levels = diff$celllines[order(diff$tissue)]) #Classified by tissue

e =  ggplot(diff, aes(x = celllines, y = erlotinib, fill = tissue))+ 
  geom_bar(stat = "identity") + 
  coord_flip() + 
  labs(title = "Mean graph plot of NLGI50 values for Erlotinib")

l = ggplot(diff, aes(x = celllines, y = lapatinib, fill = tissue)) + 
  geom_bar(stat="identity") + 
  coord_flip() + 
  labs(title="Mean graph plot of NLGI50 values for Lapatinib")

ggarrange(e, l,
          labels = c("A", "B"),
          ncol = 2, nrow = 1)
rm(e,l)

```

The difference from the NegLogGI50 for a particular cell line and the mean NegLogGI50 is plotted here for Erlotinib and Lapatinib.

### 4.2.3 Pearson correlation {#css_class4.2.3}
```{r correlation erlotinib, lapatinib}
cor(NLGI50.all$erlotinib, NLGI50.all$lapatinib, method = "pearson")
```

A Pearson correlation coefficient of ~ 0.65 confirms that these patterns are similar. One reason for this is that both erlotinib and lapatinib are EGFR inhibitors. Cell lines that were more sensitive are displayed as bars that project to the right of the mean. Cell lines that were less sensitive are displayed with bars projected to the left.

### 4.2.4 Lung cellines {#css_class4.2.4}
To answer our more specific question, whether lapatinib also has an effect on lung cancer, we will only look at the cell lines in lung cancer.

```{r lung genes, fig.height = 7, fig.width = 14}
#only lung with mean all
### load data
Metadata_Lapatinib_treated = Metadata[which(Metadata$drug == "lapatinib" & Metadata$dose != "0nM"),]
NegLogGI50 = as.data.frame(readRDS(paste0(wd, "/Data/NegLogGI50.rds")))
#lung cells from Metadata
Lung_Metadata_L_treated = Metadata[which(Metadata$drug == "lapatinib" & Metadata$dose != "0nM" & Metadata$tissue == "Lung"),] 
celllines = Lung_Metadata_L_treated$cell
NegLogGI50.lung = as.data.frame(t(NegLogGI50[c("erlotinib", "lapatinib"), celllines]))

#Difference
dif.NegLogGI50.lung = data.frame(erlotinib = NegLogGI50.lung$erlotinib -  mean(NLGI50.all$erlotinib), lapatinib = NegLogGI50.lung$lapatinib -  mean(NLGI50.all$lapatinib)) #erlotinib data - mean value, lapatinib data - mean value
dif.NegLogGI50.lung$celllines = rownames(NegLogGI50.lung)

# PLot Erlotinib and Lapatinib

e = ggplot(dif.NegLogGI50.lung,aes(x = celllines, y = erlotinib)) + geom_bar(stat = "identity", fill = "skyblue") + geom_text(aes(label=round(erlotinib, 2)), vjust = -0.5, color = "black", size = 3) + coord_flip() + labs(title = "Mean graph plot of NLGI50 values for Erlotinib, only lung genes")
l = ggplot(dif.NegLogGI50.lung,aes(x = celllines, y = lapatinib)) + geom_bar(stat = "identity", fill = "skyblue") + geom_text(aes(label=round(lapatinib, 2)), vjust = -0.5, color = "black", size = 3) + coord_flip() + labs(title = "Mean graph plot of NLGI50 values for Lapatinib, only lung genes")

ggarrange(e, l,
          labels = c("A", "B"),
          ncol = 2, nrow = 1)
rm(e,l)

```

The difference from the NegLogGI50 for cell line from lung tissue and the mean NegLogGI50 is plotted here for Erlotinib and Lapatinib.


```{r pearson correlation lung}
#correlation lung

cor(NegLogGI50.lung$erlotinib, NegLogGI50.lung$lapatinib, method = "pearson")

```

A pearson correlation coefficent of ~ 0.96 suggests that Lapatinib has a similar effect on lung cancer as Erlotinib.


<div style="text-align: justify"> 
### Selection of Lapatinib and Erlotinib treated cells
</div>
```{r}
lapa<-data.frame(Metadata[which(Metadata[,'drug'] == "lapatinib"), ])
erlo<-data.frame(Metadata[which(Metadata[,'drug'] == "erlotinib"), ])
el<-right_join(lapa,erlo, by="cell")
rmv.rows = apply(el, 1, function(x) {
  sum(is.na(x))
})  # Go through each row and sum up all missing values
row.names(rmv.rows)
```

<div style="text-align: justify"> 
Create data frame with Lapatinib and Erlotinib data </div>
```{r}
fc<-data.frame(Treated-Untreated)
all<-data.frame(fc[grep("lapatinib|erlotinib", colnames(fc))])


all.rmv<-data.frame(all %>% dplyr::select(
                    -"CCRF.CEM_erlotinib_10000nM_24h", 
                    -"HL.60_erlotinib_10000nM_24h", 
                    -"HT29_erlotinib_10000nM_24h", 
                    -"K.562_erlotinib_10000nM_24h", 
                    -"LOX_erlotinib_10000nM_24h",
                    -"SR_erlotinib_10000nM_24h",
                    -"COLO.205_lapatinib_10000nM_24h"))
                  
la<-data.frame(all.rmv[grep("lapatinib", colnames(all.rmv))])

er<-data.frame(all.rmv[grep("erlotinib", colnames(all.rmv))])
erla<-data.frame(er,la)
Drug<-c(rep('Erlotinib',53), rep('Lapatinib',53))
log2FoldChange<-apply(erla, MARGIN = 2, FUN = mean)
df_drug<-data.frame(log2FoldChange, Drug)

```

<div style="text-align: justify"> 
### 4.2.5 Anova {#css_class4.2.5}
### Checking requirements for ANOVA
First, the fold change data of lapatinib and erlotinib must be checked for normality. Anova can be used if the data is distributed normally.</div>
```{r}
qqnorm(log2FoldChange, pch = 1, frame = FALSE)
qqline(log2FoldChange, col = "steelblue", lwd = 2)
```
<div style="text-align: justify"> 
Anova anticipates homogeneity of variances. If the p-value of the Fligner-Killeen test is greater than 0.05, the homogeneity of variance is not rejected and this condition is met. The p-value of 0,08067 let us know that variances are homogeneous.</div>
```{r}
fligner.test(log2FoldChange ~ Drug, data=df_drug )
```

<div style="text-align: justify"> 
Anova tests whether the means of several independent groups differ.
A p-value of 0.53 means that the result is not significant. Therefore, there is no significant difference between Erlotinib and Lapatinib.</div>
```{r}
ggboxplot (data = df_drug, x="Drug", y="log2FoldChange", color = "Drug",    
          add = "jitter", legend = "none")+
          rotate_x_text(angle = 45)+
          stat_compare_means(method = "anova",  label.x = 2)+ # Add global anova p-value
          stat_compare_means(label = "p.signif", method = "t.test",
          ref.group = ".all.", hide.ns = TRUE)      # Pairwise comparison against all
```



## 4.3 Question 3: Does lapatinib have a similar effect on brain metastasis as it does on breast cancer? {#css_id4.3}

As a proven fact, lapatinb crosses the brain-blood barrier. Since breast cancer tends to spread and form metastasis in brain cells and lapatinib is used in anti-breast cancer therapy, we wanted to analyse our data for similar effects after lapatinib treatment in breast and cns tissue cells. 

At first, we selected lapatinib response genes by creating matrices with cns- and breast-cell line fold change values. 
```{r}
L_fc <- select(Fold_Change, contains("Lapa"))
L_fc <- as.data.frame(t(L_fc))
rownames(Metadata) <- Metadata$sample


L_treated <- select(Treated, contains("Lapa"))
L_treated <- t(L_treated)
L_untreated <- select(Untreated, contains("Lapa"))
L_untreated <- t(L_untreated)



# selecting breast Lapatinib samples
breast <- Metadata[Metadata[,'tissue']=="Breast",]
rownames(breast) <- breast$sample
rownames(breast) <- gsub(x = rownames(breast), pattern = "-", replacement = ".")  

breastFC <- subset(L_fc, rownames(L_fc) %in% rownames(breast))
breastTreated <- subset(L_treated, rownames(L_treated) %in% rownames(breast))
breastUntreated <- subset(L_untreated, rownames(L_untreated) %in% rownames(breast))#


# selecting CNS Lapatinib samples
cns <- Metadata[Metadata[,'tissue']=="CNS",]
rownames(cns) <- cns$sample
rownames(cns) <- gsub(x = rownames(cns), pattern = "-", replacement = ".")

cnsFC <- subset(L_fc, rownames(L_fc) %in% rownames(cns))
cnsTreated <- subset(L_treated, rownames(L_treated) %in% rownames(cns))
cnsUntreated <- subset(L_untreated, rownames(L_untreated) %in% rownames(cns))
```

Next, we performed a paired t-test of treated and untreated cns and breast samples, respectively, to determine statistical significant fold change values. Since we are performing multiple testing over different cell lines from breast and cns tissues, we want to adjust our p value by performing a Benjamini Hochberg correction.

```{r}
#performing a paired t-test of treated and untreated samples
t_test_cns <- col_t_paired(cnsTreated, cnsUntreated, alternative = "two.sided", mu = 0,conf.level = 0.95)
t_test_breast <- col_t_paired(breastTreated, breastUntreated, alternative = "two.sided", mu = 0,conf.level = 0.95)

#obtaining Benjamini-Hochberg adjusted p-values
pval_cns <- t_test_cns$pvalue
pval_breast <- t_test_breast$pvalue

fdr_cns <- p.adjust(pval_cns, "BH")
fdr_breast <- p.adjust(pval_breast, "BH")


#obtaining mean FC values over all samples 
breastFCm <- as.numeric(colMeans(breastFC))
cnsFCm <- as.numeric(colMeans(cnsFC))
genes <- colnames(breastFC)
```


### 4.3.1 Volcano plot {#css_class4.3.1}

For visualization of the biological significance (fold change values) and the statistical significance (adjusted p-values), we plotted our data in an interactive volcano plot.

```{r}
## breast volcano plot
#creating a matrix containg all needed values for plotting
diff_df_breast <- data.frame(gene = genes, Fold = breastFCm, FDR = fdr_breast)
diff_df_breast$absFold <- abs(diff_df_breast$Fold)
head(diff_df_breast)


# add a grouping column; default value is "not significant"
diff_df_breast$group <- "NotSignificant"



# change the grouping for the entries with significance but not a large enough Fold change
diff_df_breast[which(diff_df_breast['FDR'] < 0.5 & (diff_df_breast['absFold']) < 0.5 ),"group"] <- "Significant"

# change the grouping for the entries a large enough Fold change but not a low enough p value
diff_df_breast[which(diff_df_breast['FDR'] > 0.5 & (diff_df_breast['absFold']) > 0.5 ),"group"] <- "FoldChange"

# change the grouping for the entries with both significance and large enough fold change
diff_df_breast[which(diff_df_breast['FDR'] < 0.5 & (diff_df_breast['absFold']) > 0.5 ),"group"] <- "Significant&FoldChange"


# Find and label the top peaks.
top_peaks_breast <- diff_df_breast[with(diff_df_breast, order(Fold, FDR)),][1:10,]
top_peaks_breast <- rbind(top_peaks_breast, diff_df_breast[with(diff_df_breast, order(-Fold, FDR)),][1:10,])


# Add gene labels for all of the top genes we found
# creating an empty list, and filling it with entries for each row in the dataframe
# each list entry is another list with named items that will be used 
a <- list()
for (i in seq_len(nrow(top_peaks_breast))) {
  m <- top_peaks_breast[i, ]
  a[[i]] <- list(
    x = m[["Fold"]],
    y = -log10(m[["FDR"]]),
    text = m[["gene"]],
    xref = "x",
    yref = "y",
    showarrow = TRUE,
    arrowhead = 0.5,
    ax = 20,
    ay = -40
  )
}

plot_breast <- plot_ly(data = diff_df_breast, x = diff_df_breast$Fold, y = -log10(diff_df_breast$FDR), type = "scatter", text = diff_df_breast$gene, mode = "markers", color = diff_df_breast$group) %>% 
  layout(title ="Volcano Plot of Lapatinib breast cancer samples", 
         xaxis = list(title="log2 Fold Change"),
         yaxis = list(title="FDR")) %>%
  layout(annotations = a)
plot_breast

###thresholds still need to be discussed
```
```{r}

## CNS volcano plot

diff_df_cns <- data.frame(gene = genes, Fold = cnsFCm, FDR = fdr_cns)
diff_df_cns$absFold <- abs(diff_df_cns$Fold)
head(diff_df_cns)


# add a grouping column; default value is "not significant"
diff_df_cns$group <- "NotSignificant"


# change the grouping for the entries with significance but not a large enough Fold change
diff_df_cns[which(diff_df_cns['FDR'] < 0.5 & (diff_df_cns['absFold']) < 0.5 ),"group"] <- "Significant"

# change the grouping for the entries a large enough Fold change but not a low enough p value
diff_df_cns[which(diff_df_cns['FDR'] > 0.5 & (diff_df_cns['absFold']) > 0.5 ),"group"] <- "FoldChange"

# change the grouping for the entries with both significance and large enough fold change
diff_df_cns[which(diff_df_cns['FDR'] < 0.5 & (diff_df_cns['absFold']) > 0.5 ),"group"] <- "Significant&FoldChange"


# Find and label the top peaks..
top_peaks_cns <- diff_df_cns[with(diff_df_cns, order(Fold, FDR)),][1:10,]
top_peaks_cns <- rbind(top_peaks_cns, diff_df_cns[with(diff_df_cns, order(-Fold, FDR)),][1:10,])


a <- list()
for (i in seq_len(nrow(top_peaks_cns))) {
  m <- top_peaks_cns[i, ]
  a[[i]] <- list(
    x = m[["Fold"]],
    y = -log10(m[["FDR"]]),
    text = m[["gene"]],
    xref = "x",
    yref = "y",
    showarrow = TRUE,
    arrowhead = 0.5,
    ax = 20,
    ay = -40
  )
}

plot_cns <- plot_ly(data = diff_df_cns, x = diff_df_cns$Fold, y = -log10(diff_df_cns$FDR),type = "scatter", text = diff_df_cns$gene, mode = "markers", color = diff_df_cns$group) %>% 
  layout(title ="Volcano Plot of Lapatinib CNS cancer samples",
         xaxis = list(title="log2 Fold Change"),
         yaxis = list(title="FDR"))%>%
  layout(annotations = a)
plot_cns

```

### 4.3.2 Heat map for visualization {#css_class4.3.2}

To visualize the expression patterns of the top peak genes present in both tissue types we calculated a heat map, comparing the expression profiles of the two tissue types.

```{r}
# selecet top peak genes common in cns and breast tissue
tpb_comparison <- subset(top_peaks_breast, gene %in% top_peaks_cns$gene)
tpc_comparison <- subset(top_peaks_cns, gene %in% top_peaks_breast$gene)


# order common genes alphabetically
tpb_comparison <- tpb_comparison[order(tpb_comparison$gene),]
tpc_comparison <- tpc_comparison[order(tpc_comparison$gene),]


## creating heat map of FCs to compare values 
cor_mat <- cbind("breast" = tpb_comparison$Fold, "cns" = tpc_comparison$Fold)
rownames(cor_mat) <- tpb_comparison$gene
data <- read.delim


pheatmap(
  mat               = cor_mat,
  color             = magma(10),
  border_color      = "black",
  show_colnames     = TRUE,
  show_rownames     = TRUE,
  drop_levels       = TRUE,
  fontsize          = 14,
  main              = "Comparison:
  FC levels of cns and breast top peak genes"
)
```

### 4.3.3 Investigation of correlation {#css_class4.3.3}

For better visualization of the expression patterns of the top peak genes among breast and cns tissues we computed a heat map of the mean fold changes for direct comparison. 

```{r}
#correlation between top peak gene expression patterns in breast and cns tissues treated by lapatinib
cor_mat <- as.data.frame(cor_mat)
dif.FC.BC = data.frame(breast = cor_mat$breast -  mean(t(breastFC)), cns = cor_mat$cns -  mean(t(cnsFC))) #breast data - mean value, cns data - mean value
dif.FC.BC$gene = rownames(cor_mat)
```

To gather a more detailed conception of the gene expression regulation values (aka the fold change) we also computed a comparing bar plot showing the exact fold change values.

```{r, fig.height = 7, fig.width = 14}
# PLot

b = ggplot(dif.FC.BC,aes(x = gene, y = breast)) +
  geom_bar(stat = "identity", fill = "skyblue") + 
  geom_text(aes(label = round(breast, 2)), vjust = -0.5, color = "black", size = 3) + 
  coord_flip() + labs(title = "Mean of Fold Change for breast top peak genes")


c = ggplot(dif.FC.BC,aes(x = gene, y = cns)) +
  geom_bar(stat = "identity", fill = "skyblue") + 
  geom_text(aes(label = round(cns, 2)), vjust = -0.5, color = "black", size = 3) + 
  coord_flip() + labs(title = "Mean of Fold Change for CNS top peak genes")

ggarrange(b, c,
          labels = c("A", "B"),
          ncol = 2, nrow = 1)
rm(b,c)
```

At last, we investigated the expression regulation in both tissue types by calculating the pearson correlation between the fold change values of the top peak genes.

```{r}
#correlation
cor(cor_mat$breast, cor_mat$cns, method = "pearson")
```

The obtained correlation value of ~ 0.92 indicates that lapatinib treatment yields a similar effect on gene expression in breast and cns tissue cells and therefore might be possible to treat cancer in brain metastasis.  
</div>
